client/cmd/dexc-desktop/main.go

```diff
@@ -0,0 +1,529 @@
+		if err != nil {
+			return fmt.Errorf("error starting CPU profiler: %w", err)
+		}
+		defer pprof.StopCPUProfile()
+	}
+
+	defer func() {
+		if pv := recover(); pv != nil {
+			log.Criticalf("Uh-oh! \n\nPanic:\n\n%v\n\nStack:\n\n%v\n\n",
+				pv, string(debug.Stack()))
+		}
+	}()
+
+	// Prepare the Core.
+	clientCore, err := core.New(cfg.Core(logMaker.Logger("CORE")))
+	if err != nil {
+		return fmt.Errorf("error creating client core: %w", err)
+	}
+
+	// Handle shutdown by user (if running via terminal), or on system shutdown.
+	// TODO: SIGTERM is apparently spoofed by Go for Windows. Nice feature, but
+	// not well documented. Test to verify. Could also catch SIGKILL, which is
+	// sent after a configured timeout if the program doesn't exit on SIGTERM.
+	killChan := make(chan os.Signal, 1)
+	signal.Notify(killChan, syscall.SIGINT /* ctrl-c */, syscall.SIGTERM /* system shutdown */)
+	go func() {
+		for range killChan {
+			log.Infof("Shutting down...")
+			cancel()
+			return
+		}
+	}()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		clientCore.Run(appCtx)
+		cancel() // in the event that Run returns prematurely prior to context cancellation
+	}()
+
+	<-clientCore.Ready()
+
+	defer func() {
+		log.Info("Exiting dexc main.")
+		cancel()  // no-op with clean rpc/web server setup
+		wg.Wait() // no-op with clean setup and shutdown
+	}()
+
+	if cfg.RPCOn {
+		rpcSrv, err := rpcserver.New(cfg.RPC(clientCore, logMaker.Logger("RPC")))
+		if err != nil {
+			return fmt.Errorf("failed to create rpc server: %w", err)
+		}
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			cm := dex.NewConnectionMaster(rpcSrv)
+			err := cm.Connect(appCtx)
+			if err != nil {
+				log.Errorf("Error starting rpc server: %v", err)
+				cancel()
+				return
+			}
+			cm.Wait()
+		}()
+	}
+
+	webSrv, err := webserver.New(cfg.Web(clientCore, logMaker.Logger("WEB")))
+	if err != nil {
+		return fmt.Errorf("failed creating web server: %w", err)
+	}
+
+	webStart := make(chan error)
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		cm := dex.NewConnectionMaster(webSrv)
+		webStart <- cm.Connect(appCtx)
+		cm.Wait()
+	}()
+
+	if err := <-webStart; err != nil {
+		return err
+	}
+
+	// No errors running webserver, so we can be certain we won any race between
+	// starting instances. Start the sync server now.
+	var openC chan struct{}
+	if startServer {
+		openC = make(chan struct{}) // no buffer. Is ignored if window is currently open
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			runServer(appCtx, syncDir, openC, killChan, cfg.Net)
+		}()
+	}
+
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		systray.Run(func() {
+			systrayOnReady(appCtx, filepath.Dir(cfg.LogPath), openC, killChan)
+		}, func() {
+			wg.Done()
+		})
+	}()
+
+	openWindow := func() {
+		wg.Add(1)
+		go func() {
+			runWebviewSubprocess(appCtx, "http://"+webSrv.Addr())
+			wg.Done()
+		}()
+	}
+
+	openWindow()
+
+windowloop:
+	for {
+		var backgroundNoteSent bool
+		select {
```

Comment:
Looks like main needs a refactor so that this is in a goroutine and main is just blocking on `systray.Run`.